checker/src/options.rs

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) Facebook, Inc. and its affiliates.
+//
+// This source code is licensed under the MIT license found in the
+// LICENSE file in the root directory of this source tree.
+
+use clap::{App, AppSettings, Arg};
+use itertools::Itertools;
+use shellwords;
+
+/// Creates the clap::App metadata for argument parsing.
+fn make_options_parser<'a>() -> App<'a, 'a> {
+    // We could put this into lazy_static! with a Mutex around, but we really do not expect
+    // to construct this more then once per regular program run.
+    App::new("MIRAI")
+    .setting(AppSettings::NoBinaryName)
+    .version("v1.0.5")
+    .arg(Arg::with_name("single_func")
+        .long("single_func")
+        .takes_value(true)
+        .help("Focus analysis on the named function.")
+        .long_help("Name is the simple name of a top-level crate function or a MIRAI summary key."))
+    .arg(Arg::with_name("test_only")
+        .long("test_only")
+        .takes_value(false)
+        .help("Focus analysis on #[test] methods.")
+        .long_help("Only #[test] methods and their usage are analyzed. This must be used together with the rustc --test option."))
+    .arg(Arg::with_name("flaky")
+        .long("flaky")
+        .takes_value(false)
+        .help("Report diagnostics even with incomplete information.")
+        .long_help("This removes measurements to suppress flaky diagnostics. The effect is implementation dependent."))
+}
+
+/// Represents options passed to MIRAI.
+#[derive(Debug, Default)]
+pub struct Options {
+    pub single_func: Option<String>,
```

Comment:
May as well add the "tion".

Reply:
Its rather common to use this abbreviation. The 'fn' as well. I keep it (but have renamed fn).